jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -146,38 +155,37 @@ private void publishList(XmlUtils xmlUtils, List<OdeData> dataList) throws JsonP
 		serialId.setBundleSize(dataList.size());
 		for (OdeData odeData : dataList) {
 			OdeLogMetadata msgMetadata = (OdeLogMetadata) odeData.getMetadata();
+			OdeMsgPayload msgPayload = (OdeMsgPayload) odeData.getPayload();
 			msgMetadata.setSerialId(serialId);
 
 			if (isDriverAlertRecord()) {
 				logger.debug("Publishing a driverAlert.");
 
 				publisher.publish(JsonUtils.toJson(odeData, false),
 						publisher.getOdeProperties().getKafkaTopicDriverAlertJson());
-			} 
-			else 
-			{
+			} else {
 				if (isBsmRecord()) {
 					logger.debug("Publishing a BSM");
-				} else if(isSpatRecord()) {
+				} else if (isSpatRecord()) {
 					logger.debug("Publishing a Spat");
-				}else {
-					logger.debug("Publishing a TIM");
+				} else {
+					logger.debug("Publishing a TIM or MAP");
 				}
-				
-				if(isSpatRecord() && msgMetadata instanceof OdeSpatMetadata 
-						&& !((OdeSpatMetadata)msgMetadata).getIsCertPresent() )
-				{
-					//Nothing: If Spat log file and IEEE1609Cert is not present, Skip the Ieee1609Dot2Data encoding					
-				}
-				else 
-				{
-					Asn1Encoding msgEncoding = new Asn1Encoding("root", "Ieee1609Dot2Data", EncodingRule.COER);
-					msgMetadata.addEncoding(msgEncoding);
+
+				if (isSpatRecord() && msgMetadata instanceof OdeSpatMetadata
+						&& !((OdeSpatMetadata) msgMetadata).getIsCertPresent()) {
+					// Nothing: If Spat log file and IEEE1609Cert is not present, Skip the
```

Comment:
You didn't write this but this bugs me. An 'if' statement to run nothing. This should just be written to check the negated value of this conditional to not have an unused 'if' statement and then move the 'else' code into it.